ext/opcache/jit/zend_jit_ir.c

```diff
@@ -2702,14 +2656,18 @@ static void zend_jit_init_ctx(zend_jit_ctx *jit, uint32_t flags)
 #else /* IR_TARGET_x86 */
 			jit->ctx.fixed_stack_frame_size = sizeof(void*) * 11; /* 4 saved registers and 7 spill slots (4 bytes) */
 #endif
-			if (GCC_GLOBAL_REGS) {
-				jit->ctx.fixed_save_regset = IR_REGSET_PRESERVED & ~((1<<ZREG_FP) | (1<<ZREG_IP));
-			} else {
-				jit->ctx.fixed_save_regset = IR_REGSET_PRESERVED;
+			/* JIT-ed code is called only from execute_ex, which takes care
+			 * of saving ZREG_FP, ZREG_IP when GCC_GLOBAL_REGS is 1, so we don't
+			 * have to save them. When GCC_GLOBAL_REGS is 1, always save them.
+			 */
+#if GCC_GLOBAL_REGS
```

Comment:
Nit: This didn't actually change, but adds a bit of noise to the diff (also indent on commented code is now off)